plugins/modules/digital_ocean_droplet.py

```diff
@@ -350,20 +368,31 @@ def ensure_power_on(self, droplet_id):
             time.sleep(min(2, end_time - time.time()))
         self.module.fail_json(msg='Wait for droplet powering on timeout')
 
+    def ensure_power_off(self, droplet_id):
+        response = self.rest.post('droplets/{0}/actions'.format(droplet_id), data={'type': 'power_off'})
+        end_time = time.time() + self.wait_timeout
+        while time.time() < end_time:
+            response = self.rest.get('droplets/{0}'.format(droplet_id))
+            json_data = response.json
+            if json_data['droplet']['status'] == 'off':
+                return json_data
+            time.sleep(min(2, end_time - time.time()))
```

Comment:
This line could lead to a `ValueError` in the event of clock skew. (admittedly somewhat unlikely, but you never know on a busy machine.) For example:
```
>>> time.sleep(-1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: sleep length must be non-negative
```
I think you should just remove the subtraction here to keep it simple. It doesn't have to be exact. (maybe document that it isn't exact if you're concerned.)
Also maybe increase the sleep to 10 seconds; hitting the API every two seconds for two minutes seems a little excessive.

Reply:
These are definitely good suggestions -- it's not really apparent, but, this is a shameless copy of the function "above it" (flipping "on" to "off"). I'm going to turn this into a new issue.

Reply:
Fix in #65.